src/pu-emmc.c

```diff
@@ -656,10 +699,22 @@ pu_emmc_parse_raw(PuEmmc *emmc,
         input->uri = pu_hash_table_lookup_string(value_input->data.mapping, "uri", "");
         input->md5sum = pu_hash_table_lookup_string(value_input->data.mapping, "md5sum", "");
         input->sha256sum = pu_hash_table_lookup_string(value_input->data.mapping, "sha256sum", "");
+        if(!pu_emmc_get_input_size(input, prefix, error))
+            return FALSE;
+
         bin->input = input;
         g_debug("Parsed raw input: uri=%s md5sum=%s sha256sum=%s",
                 input->uri, input->md5sum, input->sha256sum);
 
+        PedSector min_offset = pu_emmc_get_partition_table_size(emmc);
+
+        if (bin->output_offset < min_offset) {
+            g_set_error(error, PU_ERROR, PU_ERROR_EMMC_PARSE,
+                        "Offset of raw binary is too small and would "
+                        "override partition table");
```

Comment:
`s/override/overwrite/`

The partition table would be "destroyed", hence "overwrite" is better suited here. See https://stackoverflow.com/a/8651622

Just noticed that I used the wrong term in `pu_emmc_parse_partitions()`, too. You may also change that, if you like.